### ISSUE TITLE
Make the IsVisible property a boolean

### ISSUE THREAD

**Issue by MatthieuNarcisi** [Microsoft/xaml-standard#29]
In UWP and WPF XAML, the visibility property is a value of the Visibility Enum :

```xaml
<uiElement Visibility="Visible"/>
-or-
<uiElement Visibility="Collapsed"/>
```

This is not useful anymore as the Enum now contains only two values (it used to be three).

In Xamarin.Forms it is a Boolean. This allows for direct binding to a Boolean value without the use of a converter.

We should make it so for XAML Standard.

**Comment by dotMorten** [comment]
> now contains only two values (it used to be three).

WPF still has this property, and IMHO has some value. I personally think this change would be detrimental to existing code. Also it would be a major breaking change, since IsVisible is already a (read-only) property in WPF.
UWP recently added implicitly conversion from `true` to `Visible`, and I'd rather see solutions like this over breaking the existing platform.

Having said that, I do think Xamarin.Forms should change away from IsVisible and do what two other platforms already does.

**Comment by harinikmsft** [comment]
On it! - see https://github.com/Microsoft/xaml-standard/issues/8

There are enhancements to this we are considering in UWP as well, besides from just being about toggling visibility - some perf optimizations, perhaps mapping better for expression animations on the property and maybe even extending it to do what the WPF property does (computed behavior). Let us know what you would like to see

**Comment by skendrot** [comment]
@dotMorten I doubt this new standard will "just work" with other platforms, especially WPF. Removing Visbility makes sense as **only** WPF has a third value.

**Comment by h82258652** [comment]
But in WPF, we have ```Visibility``` and ```IsVisible``` properties.

**Comment by birbilis** [comment]
btw, can't have something both be for setting wanted behaviour and for querying visual result that may be affected by a parent objec

that is because if somebody serializes the object it will get the value from the query, not the one that had been set to it originally (that defines its permanent behaviour, that kicks in say if the object gets reparented to a visible parent/tree)

so you need both Visibility and IsVisible (if you don't want to navigate the tree and also check the current view bounds and maybe things that may be overlapping [depends on what you mean with IsVisible I guess] to see if something is totally unoccluded/visible indeed)

**Comment by jassmith** [comment]
Closed as dupe of #8
